### PR TITLE
Try to upgrade ginkgo to v2

### LINKED ISSUE #3184: upgrade github.com/onsi/ginkgo to v2
Hi Fabric,

In recent I noticed ginkgo released v2 version.
Do we have any plan to do the package upgrade?
some en... benefits from the upgrade, it's vscode plugin of the ginkgo then can be used.

image

Thanks and regards
Sam

### PR DESCRIPTION
**Pull request by SamYuan1990**
#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

upgrade ginkgo to v2
upgrade related go files to ginkgo v2

#### Additional details

ref: https://onsi.github.io/ginkgo/MIGRATING_TO_V2

#### Related issues

resolve https://github.com/hyperledger/fabric/issues/3184

### CODE REVIEW

**Comment by SamYuan1990** [comment]
@denyeart , ref to https://github.com/hyperledger/fabric/pull/2954#discussion_r718875335
which means for PR related with go mod changes as https://github.com/hyperledger/fabric/pull/3202/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L24
depends on  #2707 ?

**Comment by SamYuan1990** [comment]
can someone help with rerun unit test in CI?

**Comment by SamYuan1990** [comment]
Thanks @yacovm , as CI passed, and current go mod changes without protobuf changes.
Can someone help review this PR?

File: go.mod
Comment (denyeart) [inline go.mod:24]: @guoger @Param-S I thought etcd required the older protobuf, and therefore we were stuck at the older version of protobuf and ginkgo until the etcd update is complete. Right?

Comment (Param-S) [reply go.mod:24]: Yes, It is correct. It is recorded here: https://github.com/hyperledger/fabric/pull/2185#issuecomment-737650631.

Comment (Param-S) [reply go.mod:24]: One more issue for reference: https://jira.hyperledger.org/browse/FAB-18363

File: go.mod
Comment (SamYuan1990) [inline go.mod:72]: @Param-S , @denyeart , I am not sure if we can use `replace` here to make protobuf using version 1.3.3?

Comment (SamYuan1990) [reply go.mod:72]:
```
yuanyi@yuanyideMacBook-Pro fabric % go mod why github.com/golang/protobuf                          
# github.com/golang/protobuf
(main module does not need package github.com/golang/protobuf)
```

Comment (denyeart) [reply go.mod:72]: I think you need to wait until the Protobuf update from the other PR is merged.

Comment (SamYuan1990) [reply go.mod:72]: I see

**Review by mastersingh24 — CHANGES_REQUESTED** [review]
Just blocking this until the protobuf upgrade

**Comment by SamYuan1990** [comment]
> Just blocking this until the protobuf upgrade

May I know is there any schedule for protobuf upgrade?

**Comment by SamYuan1990** [comment]
please let me know if this pr is ready for merge/review once I rebased with latest.

**Comment by SamYuan1990** [comment]
Will reopen after etcd one merged. @denyeart

**Pull request closed without merging**
